snuba/datasets/entities/factory.py

```diff
@@ -0,0 +1,57 @@
+from typing import Callable, MutableMapping, Set, Sequence
+
+from snuba import settings
+from snuba.datasets.entity import Entity
+from snuba.util import with_span
+
+ENTITY_IMPL: MutableMapping[str, Entity] = {}
+ENTITY_NAME_LOOKUP: MutableMapping[Entity, str] = {}
+
+ENTITY_NAMES: Set[str] = {
+    "outcomes",
+    "outcomes_raw",
+}
+
+ACTIVE_ENTITY_NAMES: Set[str] = ENTITY_NAMES - settings.DISABLED_ENTITIES
+
+
+class InvalidEntityError(Exception):
+    """Exception raised on invalid entity access."""
+
+
+@with_span()
+def get_entity(name: str) -> Entity:
+    if name in ENTITY_IMPL:
+        return ENTITY_IMPL[name]
+
+    if name in settings.DISABLED_ENTITIES:
+        raise InvalidEntityError(
+            f"entity {name!r} is not available in this environment"
+        )
+
+    from snuba.datasets.entities.outcomes import OutcomesEntity
+    from snuba.datasets.entities.outcomes_raw import OutcomesRawEntity
+
+    entity_factories: MutableMapping[str, Callable[[], Entity]] = {
```

Comment:
Should these keys use the enums you've defined in `__init__.py`?

Reply:
Yes I just found that bug right before standup.